File: competition/03-primes/primes.js
function isPrime(n) {
  for (let d = 2; d * d <= n; d++) {
    if (n % d === 0) return false;
  }
  return true;
}

let count = 0;
for (let n = 2; n < 300000; n++) {
  if (isPrime(n)) count++;
}
console.log(count);

File: competition/03-primes/primes.hs
-- Count the primes below a limit by trial division — integer arithmetic, modulo.
isPrime :: Int -> Bool
isPrime n = go 2
  where
    go d
      | d * d > n = True
      | n `mod` d == 0 = False
      | otherwise = go (d + 1)

main :: IO ()
main = print (length (filter isPrime [2 .. 299999]))
